grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -328,6 +329,31 @@ class QueryService implements Status {
         results as Collection<ConsentCollectionLink>
     }
 
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
+    Map<ConsentCollectionLinkDTO, List<StorageDocument>> findCollectionLinksByConsentKeyAndProjectKey(String consentKey, String projectKey) {
+        Map<ConsentCollectionLink, List<StorageDocument>> sampleInfo = new HashMap<>()
+        final session = sessionFactory.currentSession
+        final String query =
+                ' select c.id id, c.consent_key consentKey, c.project_key projectKey, c.pii pii, c.compliance compliance, c.sharing_type sharingType , c.text_sharing_type textSharingType,  ' +
+                        ' c.text_compliance textCompliance, c.require_mta requireMta, c.sample_collection_id sampleCollectionId, ' +
+                        ' sc.name collectionName, ic.summary consentName, ip.summary projectName, c.international_cohorts internationalCohorts ' +
+                        ' from consent_collection_link c ' +
+                        ' inner join issue ic on ic.project_key = c.consent_key ' +
+                        ' inner join issue ip on ip.project_key = c.project_key ' +
+                        ' left join sample_collection sc on sc.collection_id = c.sample_collection_id' +
+                        ' where c.project_key = :projectKey' +
+                        ' and c.consent_key = :consentKey'
+        List<ConsentCollectionLinkDTO> results = session.createSQLQuery(query)
+                .setResultTransformer(Transformers.aliasToBean(ConsentCollectionLinkDTO.class))
+                .setString('projectKey', projectKey)
+                .setString('consentKey', consentKey)
+                .list()
+        results.collect {
+            sampleInfo.put(it, StorageDocument.findAllByConsentCollectionLinkId(it.id))
```

Comment:
This is a nested query which will degrade performance really fast. Better to first pull the all of the ids into a single collection then query all storage documents with those ids. I also strongly recommend we do not use the GORM methods here (i.e. `StorageDocument.findAllByConsentCollectionLinkId`) ... they are notoriously slow. Much better to use raw sql to get all of the storage documents and then do something like `addEntity` over the results if you want to populate them.

Reply:
I've created a new query that receives a list of ids to get all documents, then i build the map by matching consent collections links id and storage documents id.